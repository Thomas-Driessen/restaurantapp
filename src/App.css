.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* 
* > Backgrounds
*/

.background-lightgray {
 background-color: #e6e6e6;
}

/*
* > Kitchen overview page
*/

.kitchen-overview-container {
  padding: 10px 15px;
  min-height: 75vh;
}

body, html, #root,.app, .loginPage, .container {
  width: 100%;
  height: 100%;
}


.container {
  font-size: 28px;
  color: #565656;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #FAFAFA;
}

.loginForm {
 box-sizing: border-box;
 width: 100%;
 max-width: 400px;
 padding-left: 20px;
 padding-right: 20px;
}

.inputField {
 padding-top: 16px;
}

.input {
 width: 100%;
 background: rgba(255, 255, 255, 1);
}

.submitButton {
 padding-top: 16px;
}


.btn {
 width: 100%;
 height: 55px;
 min-width: 280px;
 color: #565656;
 padding: 12px;
 font-size: 14px;
}

.box {
  width: 300px;

  border: 1.5px solid #FF6666;
  padding: 50px;
  margin: 20px;
  background-color: white;
  color: #565656;
}

.failLogInBox {
  padding: 20px;
  margin: 20px;
  background-color: white;
  color: #565656;
  text-align: center;
  font-size: 15px;
  color: #FF6666;
}