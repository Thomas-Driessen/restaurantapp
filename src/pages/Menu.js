import { Breadcrumbs } from '@material-ui/core';
import React from 'react';
import NavBar from '../components/Navigation bars/NavBar';
import SortBar from '../components/Navigation bars/SortBar';
import ProductsList from '../components/Product List/ProductsList';
import CategoryList from '../components/Product List/CategoryList';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { Button } from '@material-ui/core';

class MenuPage extends React.Component {

  constructor() {
    super();
    this.state = {
      foods: [],
      landingPageFoods: [],
      drinks: [],
      mostLiked: [],
      foodLikes: [],
      drinkLikes: [],
      mostLikedFoods: [],
      shownProducts: [],
      productType: "",
      foodCategories: [],
      drinkCategories: [],
      categoriesShown: [],
      selectedCategory: "",
      categorizedProducts: [],
      sorted: false
    };
  }

  async componentDidMount() {
    document.title = "Menu | " + this.props.name
    let mounted = true;
    await fetch(`${process.env.REACT_APP_API_URL}/api/food/available`)
      .then(res => res.json())
      .then((data) => {
        if (mounted) {
          this.setState({ foods: data })
        }
      })
        .catch((error) => {
          console.log("/api/food/available: " + error);
        })

        await fetch(`${process.env.REACT_APP_API_URL}/api/food`)
        .then(res => res.json())
        .then((data) => {
          if (mounted) {
            this.setState({ landingPageFoods: data })
          }
        })
          .catch((error) => {
            console.log("/api/food: " + error);
          })

    await fetch(`${process.env.REACT_APP_API_URL}/api/drink/available`)
      .then(res => res.json())
      .then((data) => {
        if (mounted) {
          this.setState({ drinks: data })
        }
      })
        .catch((error) => {
          console.log("/api/drink/available: " + error);
        })

    await fetch(`${process.env.REACT_APP_API_URL}/api/foodLikes`)
      .then(res => res.json())
      .then((data) => {
        if (mounted) {
          let foods = [];
          this.state.foods.map(food => {
            let element = food;
            element["likes"] = data.find(e => e.food.id === food.id).likes;
            foods.push(element);
            return true;
          })
          this.setState({ foodsLikes: foods });
          foods = [];
          this.state.landingPageFoods.map(food => {
            let element = food;
            element["likes"] = data.find(e => e.food.id === food.id).likes;
            foods.push(element);
            return true;
          })
          foods = foods.filter(item => item.onMenu === true);
          foods = foods.sort((a, b) => (a.likes < b.likes) ? 1 : ((b.likes < a.likes) ? -1 : 0));
          this.setState({ landingPageFoods: foods });
        }
      })
      .catch((error) => {
        console.log("/api/foodLikes: " + error);
      })

    await fetch(`${process.env.REACT_APP_API_URL}/api/drinkLikes`)
      .then(res => res.json())
      .then((data) => {
        if (mounted) {
          let drinks = [];
          this.state.drinks.map(drink => {
            let element = drink;
            element["likes"] = data.find(e => e.drink.id === drink.id).likes;
            drinks.push(element);
            return true;
          })
          this.setState({ drinksLikes: drinks });
        }
      })
        .catch((error) => {
          console.log("/api/drinkLikes: " + error);
        })

    await fetch(`${process.env.REACT_APP_API_URL}/api/category/food`)
      .then(res => res.json())
      .then((data) => {
        if (mounted) {
          this.setState({ foodCategories: data })
        }
      })
        .catch((error) => {
          console.log("/api/category/food: " + error);
        })

    await fetch(`${process.env.REACT_APP_API_URL}/api/category/drink`)
      .then(res => res.json())
      .then((data) => {
        if (mounted) {
          this.setState({ drinkCategories: data })
        }
      })
        .catch((error) => {
          console.log("/api/category/drink: " + error);
        })

    return () => mounted = false;
  }

  showFoods = (e) => {
    e.preventDefault();
    if (this.state.foodsLikes) {
      this.setState({ shownProducts: this.state.foodsLikes.filter(item => item.onMenu === true) });
      this.setState({ categoriesShown: this.state.foodCategories });
      this.setState({ productType: "Food" });
      this.setState({ selectedCategory: "" });
    }
  }

  showDrinks = (e) => {
    e.preventDefault();
    if (this.state.drinksLikes) {
      this.setState({ shownProducts: this.state.drinksLikes.filter(item => item.onMenu === true) });
      this.setState({ categoriesShown: this.state.drinkCategories });
      this.setState({ productType: "Drink" });
      this.setState({ selectedCategory: "" });
    }
  }

  showMostLiked = (e) => {
    e.preventDefault();
    if (this.state.sorted) {
      this.setState({ shownProducts: this.state.shownProducts.sort((a, b) => (a.id > b.id) ? 1 : ((b.id > a.id) ? -1 : 0)) });
      this.setState({ sorted: false });
    }
    else {
      this.setState({ shownProducts: this.state.shownProducts.sort((a, b) => (a.likes < b.likes) ? 1 : ((b.likes < a.likes) ? -1 : 0)) });
      this.setState({ sorted: true });
    }
  }

  selectCategory = (category) => {
    this.setState({ selectedCategory: category.categoryName });
  }

  resetCategory = (e) => {
    e.preventDefault();
    this.setState({ selectedCategory: "" });
  }

  render() {
    return (
      <div>
        <NavBar pageName="menu" />
        <SortBar showFoods={this.showFoods} showDrinks={this.showDrinks} showMostLiked={this.showMostLiked} productType={this.state.productType} />
        <Grid container spacing={1}>
          {this.state.selectedCategory !== "" ? (
            <div className="width-100-percent" style={{ padding: 15 }}>
              <Breadcrumbs aria-label="breadcrumb" className="breadcrumbs">
                <span color="inherit" className="breadcrumb breadcrumb-current">
                  Menu
                </span>
                <span color="inherit" className="breadcrumb" onClick={this.resetCategory}>
                  {this.state.productType}
                </span>
                <span className="breadcrumb breadcrumb-current" naria-current="page">
                  {this.state.selectedCategory}
                </span>
              </Breadcrumbs>
              <Grid container justify="space-between">
                <IconButton aria-label="return-to-categories" onClick={this.resetCategory} color="primary">
                  <ArrowBackIcon />
                </IconButton>
                <h1>{this.state.selectedCategory}</h1>
                <Button style={{margin: 'auto 0'}} size='medium' color={this.state.sorted === true ? 'primary.dark' : 'primary'} variant='contained' onClick={this.showMostLiked}>
                  <span>Most Liked</span>
                </Button>
              </Grid>
              <ProductsList products={this.state.shownProducts.filter(product => product.category.categoryName === this.state.selectedCategory)} productType={this.state.productType} />
            </div>
          ) : (
              this.state.categoriesShown.length ? (
                <div style={{ padding: 15 }}>
                  <Breadcrumbs aria-label="breadcrumb" className="breadcrumbs">
                    <span color="inherit" className="breadcrumb breadcrumb-current">
                      Menu
                    </span>
                    <span className="breadcrumb breadcrumb-current" naria-current="page">
                      {this.state.productType}
                    </span>
                  </Breadcrumbs>
                  <CategoryList categories={this.state.categoriesShown} selectCategory={this.selectCategory} />
                </div>
              ) : (
                  <Grid container spacing={0} style={{ padding: 0 }}>
                    <div id="homepageIntro">
                      <h1>Welcome to our restaurant!</h1>
                    </div>
                    <div id="homepageInstructionsWrapper">
                      <h2>How it works</h2>
                      <div id="homepageInstructions">
                        <div id="instructionScanQr" className="homepage-instruction">
                          <div className="homepage-instruction-content">
                            <div className="homepage-instruction-image">
                              <img src={process.env.PUBLIC_URL + "/images/qr-code.svg"} alt="QR-code" id="qrfrontpage"/>
                            </div>
                            <span className="instruction-text">Scan QR</span>
                          </div>
                        </div>
                        <div id="instructionBrowseFood" className="homepage-instruction">
                          <div className="homepage-instruction-content">
                              <div className="homepage-instruction-image">
                              <img src={process.env.PUBLIC_URL + "/images/hamburger.svg"} alt="Hamburgerfrontpage icon" id="hamburgerfrontpage"/>
                              </div>
                            <span className="instruction-text">Browse food</span>
                          </div>
                        </div>
                        <div id="instructionPay" className="homepage-instruction">
                          <div className="homepage-instruction-content">
                            <div className="homepage-instruction-image">
                              <img src={process.env.PUBLIC_URL + "/images/credit-card.svg"} alt="Creditcardfrontpage icon" id="creditcardfrontpage"/>
                            </div>
                            <span className="instruction-text">Pay orders</span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div id="foodRecommendations">
                      <h3>Prior foodlovers recommend...</h3>
                      <div className="popularfoods">
                        <div className="foodCard">
                          <img src={this.state.landingPageFoods.length ? this.state.landingPageFoods[0].image : null} alt="Pizza"/>
                          <span className="foodCardName">{this.state.landingPageFoods.length ? this.state.landingPageFoods[0].title : null}</span>
                        </div>
                        <div className="foodCard">
                          <img src={this.state.landingPageFoods.length ? this.state.landingPageFoods[1].image : null} alt="Pizza"/>
                          <span className="foodCardName">{this.state.landingPageFoods.length ? this.state.landingPageFoods[1].title : null}</span>
                        </div>
                        <div className="foodCard">
                          <img src={this.state.landingPageFoods.length ? this.state.landingPageFoods[2].image : null} alt="Pizza"/>
                          <span className="foodCardName">{this.state.landingPageFoods.length ? this.state.landingPageFoods[2].title : null}</span>
                        </div>
                        <div className="foodCard">
                          <img src={this.state.landingPageFoods.length ? this.state.landingPageFoods[3].image : null} alt="Pizza"/>
                          <span className="foodCardName">{this.state.landingPageFoods.length ? this.state.landingPageFoods[3].title : null}</span>
                        </div>
                      </div>
                    </div>
                  </Grid>
                )
            )}
        </Grid>
      </div>
    )
  }
}
export default MenuPage
