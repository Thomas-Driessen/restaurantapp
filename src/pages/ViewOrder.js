import React from 'react';
import NavBar from '../components/Navigation bars/NavBar';
import PreviousOrders from '../components/Previous Orders List/PreviousOrders'
import CurrentOrder from '../components/Current Order/CurrentOrder'
import { Button } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import currentFoodList from '../components/Current Order/CurrentFoodList'
import currentDrinkList from '../components/Current Order/CurrentDrinkList'
import tableId from '../components/TableId'

class ViewOrder extends React.Component {
  constructor() {
    super();
    this.state = { tableNumber: 1,
      updatePastOrders: 0
    };
  }

  sendOrder(){
    let order = [];
    currentDrinkList.map(currentDrink => {
        var drink ={
            tableId: tableId,
            paid: false,
            drinkId: currentDrink.id
        };
        let item = {
            title: currentDrink.title
        }
        order.push(item);
    fetch('/api/orderdrink', {
        method: 'POST',
        mode: 'cors',
        headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
    },
    body: JSON.stringify(drink)
    })
    return "Succes";
    })

    currentFoodList.map(currentFood => {
        var food ={
            tableId: tableId,
            paid: false,
            foodId: currentFood.id
        };
        let item = {
            title: currentFood.title
        }
        order.push(item);
    fetch('/api/orderfood', {
        method: 'POST',
        mode: 'cors',
        headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
    },
    body: JSON.stringify(food)
    })
    return "Succes";
    })

    currentDrinkList.splice(0,currentDrinkList.length);
    currentFoodList.splice(0,currentFoodList.length);
    sessionStorage.removeItem("currentDrinkList");
    sessionStorage.removeItem("currentFoodList");
    
    fetch('/sendorder', {
        method: 'POST',
        mode: 'cors',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(order)
    })
    
    this.setState({ previousOrderNumbers: this.state.previousOrderNumbers + 1 })  
  }

  render(){
  return(
      <div>
          <NavBar/>
          <Grid container spacing={0} style={{padding: 25}}>
            <Grid item xs={12} sm={6} lg={4} xl={3}>
                <Button color="secondary"variant="outlined">
                  Pay for orders
                </Button>
              <CurrentOrder sendOrder={ () => this.sendOrder()}/>
              <PreviousOrders previousOrderNumbers={this.state.previousOrderNumbers}/>
            </Grid>
          </Grid>
      </div>
  )};
}
export default ViewOrder