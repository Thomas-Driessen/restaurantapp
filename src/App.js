import React from 'react';
import NavBar from './components/NavBar';
import './App.css';
import ProductsList from './components/ProductsList';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#f05545',
      main: '#b71c1c',
      dark: '#7f0000',
      contrastText: '#fff',
    },
    secondary: {
      light: '#ff7961',
      main: '#f44336',
      dark: '#ba000d',
      contrastText: '#000',
    },
  },
});
function App() {
  return (
    <div className="App">
      <MuiThemeProvider theme={theme}>
      <NavBar />
      <ProductsList />
      </MuiThemeProvider>
    </div>
  );
}

export default App;
