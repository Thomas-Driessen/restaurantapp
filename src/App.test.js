import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import Menu from './pages/Menu'
import ViewOrder from './pages/ViewOrder'

test('404 Not Found', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/It seems that this page does not exist/i);
  expect(linkElement).toBeInTheDocument();
});

test('Render Menu', () => {
  const { getByText } = render(
    <Menu />
  );
  const linkElement = getByText(/Choose between Foods, Drinks and Most Liked to see our products/i);
  expect(linkElement).toBeInTheDocument();
});

test('Render ViewOrder', () => {
  const { getByText } = render(
    <ViewOrder />
  );
  const linkElement = getByText(/Pay for orders/i);
  expect(linkElement).toBeInTheDocument();
});