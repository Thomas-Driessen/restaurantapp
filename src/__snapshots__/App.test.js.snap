// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation0"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
          text-align="center"
        >
          Restaurant
        </h5>
        <div
          className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
        >
          <div
            style={
              Object {
                "float": "right",
              }
            }
          >
            <div
              aria-label="text primary button group"
              className="MuiButtonGroup-root"
              role="group"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  Menu
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButton-colorInherit MuiButton-textSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <span
                    className="MuiButton-startIcon MuiButton-iconSizeLarge"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18.06 22.99h1.66c.84 0 1.53-.64 1.63-1.46L23 5.05h-5V1h-1.97v4.05h-4.97l.3 2.34c1.71.47 3.31 1.32 4.27 2.26 1.44 1.42 2.43 2.89 2.43 5.29v8.05zM1 21.99V21h15.03v.99c0 .55-.45 1-1.01 1H2.01c-.56 0-1.01-.45-1.01-1zm15.03-7c0-8-15.03-8-15.03 0h15.03zM1.02 17h15v2h-15z"
                      />
                    </svg>
                  </span>
                  My orders 
                  
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation4"
    >
      <div
        className="MuiContainer-root MuiContainer-disableGutters MuiContainer-maxWidthLg"
      >
        <div
          aria-label="text primary button group"
          className="MuiButtonGroup-root MuiButtonGroup-fullWidth"
          role="group"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Food
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Drinks
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedText MuiButtonGroup-groupedTextHorizontal MuiButtonGroup-groupedTextPrimary MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Most Liked
            </span>
          </button>
        </div>
      </div>
    </header>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
    style={
      Object {
        "padding": 12,
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-displayBlock"
    >
      Choose between Foods, Drinks and Most Liked to see our products
    </h6>
  </div>
</div>
`;
